modules/operserv/readonly.c

```diff
@@ -55,6 +62,12 @@ os_cmd_readonly(struct sourceinfo *si, int parc, char *parv[])
 		wallops("\2%s\2 unset the READONLY option.", get_oper_name(si));
 		logcommand(si, CMDLOG_ADMIN, "READONLY:OFF");
 		command_success_nodata(si, _("Read-only mode is now disabled."));
+
+		if (commit_interval_timer)
+			(void) mowgli_timer_destroy(base_eventloop, commit_interval_timer);
+
+		commit_interval_timer = mowgli_timer_add(base_eventloop, "db_save_periodic", &db_save_periodic, NULL,
+		                                         config_options.commit_interval);
```

Comment:
This doesn't check if `db_save` is set, thus scheduling a null pointer dereference if running without a write-capable backend.

Arguably, the `READONLY` command ought to fail early in that case (similar to the check added in `operserv/set_commitinterval`) – there's little point toggling database writes when they are impossible anyway.